### ISSUE TITLE
Implement web page -> background script communication, with Firefox

### ISSUE THREAD

**Issue by fregante** [pixiebrix/extension-messaging#6]
- Follows https://github.com/pixiebrix/extension-messaging/issues/5
- More context in https://github.com/pixiebrix/pixiebrix-extension/issues/1015
- More context in https://github.com/pixiebrix/pixiebrix-extension/pull/1019

**Comment by fregante** [comment]
A few notes regarding the unsafe communication in general:

- This API will need to accept an extension ID (for `externally_connectable`), which could be defined in `getMethod("GET_VERSION", id)` (since it's static) or at the call site, so `getVersion(id)`
- The unsafe context should probably not have access to arbitrary tabs or contexts, so `getVersion({extensionId, tabId, frameId})` should not be supported
- Since this is now being built as a public module, the unsafe messaging should probably always indicate which extension it's trying to talk to, even if the content script doesn't require the extension ID.

This leaves only 3 contexts accessible from the unsafe context:

- current CS
- background context (either via `externally_connectable` or via CS)
- perhaps the current tab’s editor/devtools (either directly or via CS+background)

Since the unsafe context is special, security-wise, it should probably have its own special methods for each context anyway.

General notes also added in https://github.com/pixiebrix/webext-messenger/issues/1

**Comment by twschiller** [comment]
I agree with everything above. As you mention, the external -> devtools isn't strictly necessary because it can hop through the background page. (I don't think CS can initiate communication to the devtools in our current setup?)

Also commented in https://github.com/pixiebrix/webext-messenger/issues/5#issuecomment-922568232 about opt-in for endpoints
